manifests/install/archive.pp

```diff
@@ -0,0 +1,90 @@
+# Class: zookeeper::install::archive
+#
+# This module manages Zookeeper installation from an archive file supported by
+# the 'archive' type
+#
+# Parameters: None
+#
+# Actions: None
+#
+#
+# Should not be included directly
+#
+class zookeeper::install::archive(
+  $mirror_url        = $::zookeeper::install::mirror_url,
+  $install_dir       = $::zookeeper::install::zoo_dir,
+  $archive_checksum  = $::zookeeper::install::archive_checksum,
+  $package_dir       = $::zookeeper::install::package_dir,
+  $ensure            = $::zookeeper::install::ensure,
+  $snap_retain_count = $::zookeeper::install::snap_retain_count,
+  $cleanup_sh        = $::zookeeper::install::cleanup_sh,
+  $datastore         = $::zookeeper::install::datastore,
+  $user              = $::zookeeper::install::user,
+  $group             = $::zookeeper::install::group,
+  $ensure_account    = $::zookeeper::install::ensure_account,
+  $service_provider  = $::zookeeper::install::service_provider,
+  $ensure_cron       = $::zookeeper::install::ensure_cron,
+  $service_package   = $::zookeeper::install::service_package,
+  $packages          = $::zookeeper::install::packages,
+  $cdhver            = $::zookeeper::install::cdhver,
+  $install_java      = $::zookeeper::install::install_java,
+  $java_package      = $::zookeeper::install::java_package,
+  $repo              = $::zookeeper::install::repo,
+  $manual_clean      = $::zookeeper::install::manual_clean,
+) {
+
+  include '::archive'
+
+  $basefilename = "zookeeper-${ensure}.tar.gz"
+  $package_url = "${mirror_url}/zookeeper/zookeeper-${ensure}/${basefilename}"
```

Comment:
By default this would have value `zookeeper-present`, which would throw an error. And that's ok, maybe we would like to use some more reasonable value... like http://mirror.cogentco.com/pub/apache/zookeeper/stable/ but still you would need to know the exact version...